engine/dl_list.cpp

```diff
@@ -143,16 +143,20 @@ bool DLList::Replace(DLRecord* old_record, DLRecord* new_record,
       old_record->PersistPrevNT(new_record_offset);
     } else {
       new_record->prev = prev_offset;
+#ifdef KVDK_WITH_PMEM
```

Comment:
Maybe we should redefine pmem_persist()/pmem_memcpy()/pmem_memmove() as  macros for compiling without pmem so that we can reduce macros?

Reply:
Good suggestion. I find `7` locations where checking `KVDK_WITH_PMEM` could be removed with this improvement.